File: frontend/src/services/analytics.ts
import { api } from './api';

export interface DashboardOverview {
  total_users: number;
  total_products: number;
  total_orders: number;
  total_revenue: number;
}

export interface MonthlySale {
  month: string;
  sales: number;
  revenue: number;
}

export interface CategoryAnalytic {
  category: string;
  total_sales: number;
  revenue: number;
}

export interface DemandForecastPoint {
  day: string;
  orders: number;
}

export const analyticsService = {
  async getDashboardOverview(): Promise<DashboardOverview> {
    const res = await api.get<DashboardOverview>('/api/analytics/dashboard');
    return res.data;
  },

  async getSalesAnalytics(): Promise<any> {
    const res = await api.get('/api/analytics/sales');
    return res.data;
  },

  async getRevenueAnalytics(): Promise<any> {
    const res = await api.get('/api/analytics/revenue');
    return res.data;
  },

  async getMonthlySales(): Promise<MonthlySale[]> {
    const res = await api.get<MonthlySale[]>('/api/analytics/monthly-sales');
    return res.data;
  },

  async getTopSellingProducts(limit = 10): Promise<any[]> {
    const res = await api.get<any[]>('/api/analytics/top-products', {
      params: { limit }
    });
    return res.data;
  },

  async getCategoryAnalytics(): Promise<CategoryAnalytic[]> {
    const res = await api.get<CategoryAnalytic[]>('/api/analytics/categories');
    return res.data;
  },

  async getUserAnalytics(): Promise<any> {
    const res = await api.get('/api/analytics/users');
    return res.data;
  },

  async getCustomerActivity(): Promise<any> {
    const res = await api.get('/api/analytics/customer-activity');
    return res.data;
  },

  async getOrderAnalytics(): Promise<any> {
    const res = await api.get('/api/analytics/orders');
    return res.data;
  },

  async getReviewAnalytics(): Promise<any> {
    const res = await api.get('/api/analytics/reviews');
    return res.data;
  },

  async getSentimentStatistics(): Promise<any> {
    const res = await api.get('/api/analytics/sentiment');
    return res.data;
  },

  async getFakeReviewStatistics(): Promise<any> {
    const res = await api.get('/api/analytics/fake-reviews');
    return res.data;
  },

  async getRecommendationAnalytics(): Promise<any> {
    const res = await api.get('/api/analytics/recommendations');
    return res.data;
  },

  async getModelPerformance(): Promise<any> {
    const res = await api.get('/api/analytics/model-performance');
    return res.data;
  },

  async getInventoryAnalytics(): Promise<any> {
    const res = await api.get('/api/analytics/inventory');
    return res.data;
  },

  async getDemandForecast(): Promise<DemandForecastPoint[]> {
    const res = await api.get<DemandForecastPoint[]>('/api/analytics/demand-forecast');
    return res.data;
  },

  async getPricePredictionAnalytics(): Promise<any> {
    const res = await api.get('/api/analytics/price-prediction');
    return res.data;
  },

  async getCompleteReport(): Promise<any> {
    const res = await api.get('/api/analytics/report');
    return res.data;
  }
};

File: backend/app/api/analytics.py
from fastapi import APIRouter
from app.services.analytics_service import AnalyticsService

router = APIRouter()

analytics_service = AnalyticsService()


# ---------------- Dashboard Overview ----------------
@router.get("/dashboard")
def dashboard_overview():
    return analytics_service.dashboard_overview()


# ---------------- Sales Analytics ----------------
@router.get("/sales")
def sales_analytics():
    return analytics_service.sales_analytics()


# ---------------- Revenue Analytics ----------------
@router.get("/revenue")
def revenue_analytics():
    return analytics_service.revenue_analytics()


# ---------------- Monthly Sales ----------------
@router.get("/monthly-sales")
def monthly_sales():
    return analytics_service.monthly_sales()


# ---------------- Top Selling Products ----------------
@router.get("/top-products")
def top_products(limit: int = 10):
    return analytics_service.top_products(limit)


# ---------------- Category Analytics ----------------
@router.get("/categories")
def category_analytics():
    return analytics_service.category_analytics()


# ---------------- User Analytics ----------------
@router.get("/users")
def user_analytics():
    return analytics_service.user_analytics()


# ---------------- Customer Activity ----------------
@router.get("/customer-activity")
def customer_activity():
    return analytics_service.customer_activity()


# ---------------- Order Analytics ----------------
@router.get("/orders")
def order_analytics():
    return analytics_service.order_analytics()


# ---------------- Review Analytics ----------------
@router.get("/reviews")
def review_analytics():
    return analytics_service.review_analytics()


# ---------------- Sentiment Analysis Statistics ----------------
@router.get("/sentiment")
def sentiment_statistics():
    return analytics_service.sentiment_statistics()


# ---------------- Fake Review Statistics ----------------
@router.get("/fake-reviews")
def fake_review_statistics():
    return analytics_service.fake_review_statistics()


# ---------------- Recommendation Analytics ----------------
@router.get("/recommendations")
def recommendation_analytics():
    return analytics_service.recommendation_analytics()


# ---------------- AI Model Metrics ----------------
@router.get("/model-performance")
def model_performance():
    return analytics_service.model_performance()


# ---------------- Inventory Analytics ----------------
@router.get("/inventory")
def inventory_analytics():
    return analytics_service.inventory_analytics()


# ---------------- Demand Forecast Analytics ----------------
@router.get("/demand-forecast")
def demand_forecast():
    return analytics_service.demand_forecast()


# ---------------- Price Prediction Analytics ----------------
@router.get("/price-prediction")
def price_prediction():
    return analytics_service.price_prediction()


# ---------------- Complete Report ----------------
@router.get("/report")
def complete_report():
    return analytics_service.complete_report()